lib/ui/painting.dart

```diff
@@ -1558,21 +1586,40 @@ class Path extends NativeFieldWrapperClass2 {
 
   /// Adds a new subpath that consists of the given path offset by the given
   /// offset.
-  void addPath(Path path, Offset offset) {
+  /// 
+  /// If matrix4 is specified, the path will be transformed by this matrix
+  /// (after the matrix is translated by the given offset).
+  void addPath(Path path, Offset offset, {Float64List matrix4}) {
     assert(path != null); // path is checked on the engine side
     assert(_offsetIsValid(offset));
-    _addPath(path, offset.dx, offset.dy);
+    
+    if (matrix4 != null) {
+      assert(_matrix4IsValid(matrix4));
+      _addPathWithMatrix(path, offset.dx, offset.dy, matrix4);
+    } else {
+      _addPath(path, offset.dx, offset.dy);
+    }
   }
   void _addPath(Path path, double dx, double dy) native 'Path_addPath';
-
+  void _addPathWithMatrix(Path path, double dx, double dy, Float64List matrix) native 'Path_addPathWithMatrix';
+  
   /// Adds the given path to this path by extending the current segment of this
   /// path with the the first segment of the given path.
-  void extendWithPath(Path path, Offset offset) {
+  /// 
+  /// If matrix4 is specified, the path will be transformed by this matrix
+  /// (after the matrix is translated by the given offset).
+  void extendWithPath(Path path, Offset offset, {Float64List matrix4}) {
     assert(path != null); // path is checked on the engine side
     assert(_offsetIsValid(offset));
+
+    if (matrix4 != null) {
+      assert(_matrix4IsValid(matrix4));
+      _extendWithPathAndMatrix(path, offset.dx, offset.dy, matrix4);
+    }
     _extendWithPath(path, offset.dx, offset.dy);
```

Comment:
given that you've added a way to enumerate the path, this would be a perfect candidate for something to test. :-)

Reply:
The enumeration is only possible if there are multiple disconnected segments.  That could certainly be tested.  Am I right in believing those tests would go in flutter/flutter?

Reply:
What I meant was that the extendWithPath method would be a good candidate to test, e.g. here you would have tested calling the method with a path that has two segments, and given a matrix, and verified that the result had the right number of segments -- without the return or else, it wouldn't).

But yes, everything in general should be tested. For this kind of thing, the appropriate place is:
https://github.com/flutter/engine/tree/master/testing/dart